clients/da-vinci-client/src/main/java/com/linkedin/davinci/config/VeniceServerConfig.java

```diff
@@ -760,9 +760,11 @@ public VeniceServerConfig(VeniceProperties serverProperties, Map<String, Map<Str
         serverProperties.getInt(SERVER_NON_EXISTING_TOPIC_INGESTION_TASK_KILL_THRESHOLD_SECOND, 15 * 60); // 15 mins
     nonExistingTopicCheckRetryIntervalSecond =
         serverProperties.getInt(SERVER_NON_EXISTING_TOPIC_CHECK_RETRY_INTERNAL_SECOND, 60); // 1min
-    leaderCompleteStateCheckEnabled = serverProperties.getBoolean(SERVER_LEADER_COMPLETE_STATE_CHECK_ENABLED, true);
+    leaderCompleteStateCheckEnabled =
+        serverProperties.getBoolean(SERVER_LEADER_COMPLETE_STATE_CHECK_IN_FOLLOWER_ENABLED, true);
+    /** As this is used to ignore the stale heartbeats, defaulting to 5 minutes, inline with {@link SERVER_PROMOTION_TO_LEADER_REPLICA_DELAY_SECONDS} */
     leaderCompleteStateCheckValidIntervalMs =
```

Comment:
In theory, these two things are not related; my understanding is that when (1) leader check is enabled and (2) follower hasn't received the first HB message from leader, follower doesn't even use this valid interval config; therefore, even though leader takes 30 minutes to start acting as leader, follower will not go into COMPLETE state until it sees the first HB.

However, the leader promotion delay will matter if we decide to pick [option 3 from the other issue being discussed](https://github.com/linkedin/venice/pull/775#discussion_r1419741991), and reuse this invalid interval config for that purpose.